### PR TITLE
Swagger Changes to Support Extended Groups for NFSv3

### PR DESCRIPTION
**Pull request by cagrawal10**
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

### CODE REVIEW

**Comment by cagrawal-msft** [comment]
Following files have been changed in this PR, other files are same as older versions.

/main/specification/storage/resource-manager/Microsoft.Storage/stable/2023-04-01  is picked from:
https://msazure.visualstudio.com/One/_git/Storage-SRP?path=/src/Swagger/Microsoft.Storage/stable/2023-04-01&version=GBmaster&_a=contents

/main/specification/storage/resource-manager/Microsoft.Storage/stable/2023-10-01/storage.json

/main/specification/storage/resource-manager/Microsoft.Storage/stable/2023-10-01/examples/LocalUserCreate.json
/main/specification/storage/resource-manager/Microsoft.Storage/stable/2023-10-01/examples/LocalUserGet.json
/main/specification/storage/resource-manager/Microsoft.Storage/stable/2023-10-01/examples/LocalUserUpdate.json

New files
/main/specification/storage/resource-manager/Microsoft.Storage/stable/2023-10-01/examples/LocalUsersListNFSv3Enabled.json

**Comment by blueww** [comment]
@cagrawal-msft 

Please split the commit as following, or the PR is very difficult to review.

- 1 Commit to copy from old API version (without any change)
- 1 commit to change API version.
- following commits for API change of the features.

See example : https://github.com/Azure/azure-rest-api-specs/pull/23918/commits

**Comment by cagrawal-msft** [comment]
@blueww I have added commits as you asked. Can you review it now? Let me know if anything else is required from my side.

**Comment by blueww** [comment]
> @blueww I have added commits as you asked. Can you review it now? Let me know if anything else is required from my side.

As I see, the commit in the PR is still not organized as my comments.
The new SRP API version swagger must based on latest public SRP API version , in this case it's [azure-rest-api-specs/specification/storage/resource-manager/Microsoft.Storage/stable/2023-01-01 at main · Azure/azure-rest-api-specs (github.com)](https://github.com/Azure/azure-rest-api-specs/tree/main/specification/storage/resource-manager/Microsoft.Storage/stable/2023-01-01)
You need copy from the latest public API version swagger to the API version folder, make a commit, then change the API version to new, make a commit, then add you swagger change. Without this commit origination, it's very hard to review the swagger PR.

**Comment by blueww** [comment]
Besides that, I see you added 2 API version (2023-04-01, 2023-10-01):
1. Which API version would you like client tool be cased on? You need change [Readme.md files](https://github.com/Azure/azure-rest-api-specs/blob/personal/cagrawal/ExtendedGroupSupportNFSv3/specification/storage/resource-manager) add a new tag for a new API version, and specify which is the default API version. The client tool will develop with default API version.
2. Normally we will only add at most 1 API version with 1 swagger PR.

The PR is much different from a normal SRP swagger PR for new API version. SRP team owns swagger. You can reference the before PRs or contact the creator of these PRs to get more details how to build a good SRP swagger PR for new API version: 
https://github.com/Azure/azure-rest-api-specs/pull/21500
https://github.com/Azure/azure-rest-api-specs/pull/23918
https://github.com/Azure/azure-rest-api-specs/pull/19357

File: specification/storage/resource-manager/Microsoft.Storage/stable/2023-10-01/storage.json [hunk 1]
```diff
@@ -0,0 +1,6041 @@
...
+      "get": {
+        "tags": [
+          "LocalUsers"
+        ],
+        "operationId": "LocalUsers_List",
+        "description": "List the local users associated with the storage account.",
+        "x-ms-examples": {
+          "ListLocalUsers": {
+            "$ref": "./examples/LocalUsersList.json"
+          }
```
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2023-10-01/storage.json:2000]: missed comma between examples.
```suggestion
          },
```

File: specification/storage/resource-manager/Microsoft.Storage/stable/2023-04-01/storagetasks.json [hunk 2]
```diff
@@ -0,0 +1,863 @@
+{
```
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2023-04-01/storagetasks.json:1]: Why the file is here？
As I Storage Task has changed name to Storage Action, and added to a new RP: https://github.com/Azure/azure-rest-api-specs/tree/main/specification/storageactions/resource-manager

**Pull request closed without merging**